File: src/test/java/com/melonbase/microquark/repo/CantonTest.java
package com.melonbase.microquark.repo;

import com.melonbase.microquark.repo.data.Canton;
import org.junit.jupiter.api.Test;

import java.util.stream.Stream;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class CantonTest {

  @Test
  void correctNumberOfCantons() {
    assertEquals(26, Canton.values().length);
  }

  @Test
  void correctTotalInhabitants() {
    assertEquals(8_551_744, Stream.of(Canton.values())
        .mapToLong(canton -> canton.getInhabitants())
        .sum());
  }
}

File: src/test/code/com/melonbase/microquark/repo/CantonTest.kt
package com.melonbase.microquark.repo

import com.melonbase.microquark.repo.data.Canton
import kotlin.test.Test
import kotlin.test.assertEquals

class CantonTest {

  @Test
  internal fun correctNumberOfCantons() {
    assertEquals(26, Canton.values().size)
  }

  @Test
  internal fun correctTotalInhabitants() {
    assertEquals(8_551_744, Canton.values().sumOf { it.inhabitants })
  }
}